libpromises/exec_tools.c

```diff
@@ -155,7 +155,7 @@ void ActAsDaemon()
                 GetErrorStr());
         }
 
-        if (dup2(fd, STDERR_FILENO) == -1);
+        if (dup2(fd, STDERR_FILENO) == -1)
```

Comment:
Man, I can't believe the compiler doesn't warn about this. Maybe there's a warning we can turn on deep down in the gcc argument list.